openml/extensions/sklearn/extension.py

```diff
@@ -1646,10 +1674,17 @@ def _prediction_to_probabilities(y: np.ndarray, classes: List[Any]) -> np.ndarra
                     # then we need to add a column full of zeros into the probabilities
                     # for class 3 because the rest of the library expects that the
                     # probabilities are ordered the same way as the classes are ordered).
-                    proba_y_new = np.zeros((proba_y.shape[0], len(task.class_labels)))
+
+                    # DataFrame allows more accurate mapping of classes as column names
+                    proba_y_new = pd.DataFrame(
+                        0,
+                        index=np.arange(proba_y.shape[0]),
+                        columns=task.class_labels,
+                        dtype=np.float32,
+                    )
                     for idx, model_class in enumerate(model_classes):
-                        proba_y_new[:, model_class] = proba_y[:, idx]
-                    proba_y = proba_y_new
+                        proba_y_new.loc[:, model_class] = proba_y[:, idx]
+                    proba_y = proba_y_new.to_numpy()
 
                 if proba_y.shape[1] != len(task.class_labels):
```

Comment:
This is a repeated clause from Line#1666.
If it was not true there (they are equal), then it is not true here.
If it was true there (they were unequal), then Line#1682 specifically creates `proba_y_new` to have the columns `task.class_labels`, and `proba_y` is later assigned the value of `proba_y_new`.
So I see no scenario where this statement is true (i.e. at this point `proba_y` should always a column for each label in `task.class_labels`. Am I missing something? Or is it forgotten code?

Reply:
Yes, you're right. I was scratching my head too. I think it makes sense to remove the redundant `if` but retain the warning message that is being logged.

Reply:
Sounds good to me.